smart_open/gcs.py

```diff
@@ -0,0 +1,521 @@
+# -*- coding: utf-8 -*-
+"""Implements file-like objects for reading and writing to/from GCS."""
+
+import io
+import logging
+import sys
+
+import google.cloud.exceptions
+import google.cloud.storage
+import google.auth.transport.requests as google_requests
+import six
+
+import smart_open.bytebuffer
+
+logger = logging.getLogger(__name__)
+
+_READ_BINARY = 'rb'
+_WRITE_BINARY = 'wb'
+_MODES = (_READ_BINARY, _WRITE_BINARY)
+"""Allowed I/O modes for working with GCS."""
```

Comment:
Why are these docstrings? I'd expect a comment *above* each variable, instead of a docstring *below* it.

Reply:
Sphinx picks them up that way.

Not that we use Sphinx for this project, but it's a convention.

Reply:
This was done in response to @mpenkov's [comment](https://github.com/RaRe-Technologies/smart_open/pull/404#discussion_r363068746). The convention used in the S3 module is docstrings below the constant.

Reply:
OK, but I find that thoroughly confusing. In the following block of code, which comment relates to which variable?

```python

START = 0
"""Seek to the absolute start of a GCS file"""
CURRENT = 1
"""Seek relative to the current positive of a GCS file"""
END = 2
"""Seek relative to the end of a GCS file"""
_WHENCE_CHOICES = (START, CURRENT, END)

```